modules/aws/asg.go

```diff
@@ -1,12 +1,57 @@
 package aws
 
 import (
+	"fmt"
 	"testing"
+	"time"
 
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/stretchr/testify/require"
+
+	"github.com/gruntwork-io/terratest/modules/logger"
+	"github.com/gruntwork-io/terratest/modules/retry"
 )
 
+type AsgCapacityInfo struct {
+	MinCapacity     int64
+	MaxCapacity     int64
+	CurrentCapacity int64
+	DesiredCapacity int64
+}
+
+// GetCapacityInfoForAsg returns the capacity info for the queried asg as a struct, AsgCapacityInfo.
+func GetCapacityInfoForAsg(t *testing.T, asgName string, awsRegion string) AsgCapacityInfo {
+	capacityInfo, err := GetCapacityInfoForAsgE(t, asgName, awsRegion)
+	require.NoError(t, err)
+	return capacityInfo
+}
+
+// GetCapacityInfoForAsgE returns the capacity info for the queried asg as a struct, AsgCapacityInfo.
+func GetCapacityInfoForAsgE(t *testing.T, asgName string, awsRegion string) (AsgCapacityInfo, error) {
+	asgClient, err := NewAsgClientE(t, awsRegion)
+	if err != nil {
+		return AsgCapacityInfo{}, err
+	}
+
+	input := autoscaling.DescribeAutoScalingGroupsInput{AutoScalingGroupNames: []*string{aws.String(asgName)}}
+	output, err := asgClient.DescribeAutoScalingGroups(&input)
+	if err != nil {
+		return AsgCapacityInfo{}, err
+	}
+	groups := output.AutoScalingGroups
+	if len(groups) == 0 {
```

Comment:
Is it possible `len(groups) > 1`?

Reply:
No, because we only request one AutoScalingGroupName and the ASG name is unique per region.